Fundamentals Part 2: Additional resources nesting, clarity, addition (#26319)
TheOdinProject/curriculum: foundations/javascript_basics/fundamentals-2.md

This section contains helpful links to other content. It isn't required, so consider it supplemental.

> - [Regular expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions), commonly known as regex, is a tool that matches or locates patterns in strings for string validation. However, be [cautious](https://softwareengineering.stackexchange.com/questions/113237/when-you-should-not-use-regular-expressions) about relying on regex too heavily at this stage. There are other various methods to process strings, and regex can be slower in comparison. Regardless, familiarizing yourself with some of its [capabilities](https://www.freecodecamp.org/news/a-quick-and-simple-guide-to-javascript-regular-expressions-48b46a68df29/) can make it a worthwhile addition to your arsenal.

Reviewer: ```suggestion
- [Regular expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions), commonly known as regex, is a tool that matches or locates patterns in strings for string validation. However, it's important to know [when you shouldn't use regular expressions](https://softwareengineering.stackexchange.com/questions/113237/when-you-should-not-use-regular-expressions). There are other various methods to process strings, and regex can be slower in comparison.
```

  - [Web Dev Simplified's Regular Expressions In 20 Minutes](https://www.youtube.com/watch?v=rhzKDrUiJVk)

>   - [The Net Ninja's Regular Expressions Tutorial](https://www.youtube.com/playlist?list=PL4cUxeGkcC9g6m_6Sld9Q4jzqdqHd2HiD)

Reviewer: Between these 2 videos, the new link that was added above, and the MDN link, I don't think we need so many resources about a subject we don't really go too in depth with. Letting users know about regex and when to avoid it I personally think would be enough at this point, and users can investigate further themselves if they want.

I'd say if anything, maybe we keep one of these videos here.

Author: Appreciate the input. Everything looks great and worth the update. I've pushed the requests and updated the repo.
